File: apps/admin_backend/src/modules/inventory/entities/batch.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToOne,
  JoinColumn,
} from 'typeorm';
import { Tenant } from '../../tenant/entities/tenant.entity';
import { Insumo } from './insumo.entity';

@Entity('batches')
export class Batch {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  tenant_id: string;

  @ManyToOne(() => Tenant)
  @JoinColumn({ name: 'tenant_id' })
  tenant: Tenant;

  @Column()
  insumo_id: string;

  @ManyToOne(() => Insumo, (insumo) => insumo.conversions)
  @JoinColumn({ name: 'insumo_id' })
  insumo: Insumo;

  @Column()
  batch_number: string;

  @Column({ type: 'date' })
  expiration_date: Date;

  @Column('decimal', { precision: 12, scale: 4 })
  remaining_stock: number;

  @Column('decimal', { precision: 12, scale: 2 })
  cost: number;

  @CreateDateColumn()
  created_at: Date;

  @UpdateDateColumn()
  updated_at: Date;
}

File: apps/pos_app/lib/data/models/inventory/batch_entity.dart
import 'package:floor/floor.dart';

@Entity(tableName: 'batches')
class BatchEntity {
  @primaryKey
  final String id;
  @ColumnInfo(name: 'insumo_id')
  final String insumoId;
  @ColumnInfo(name: 'batch_number')
  final String batchNumber;
  @ColumnInfo(name: 'received_date')
  final String? receivedDate;
  @ColumnInfo(name: 'expiration_date')
  final String expirationDate;
  @ColumnInfo(name: 'remaining_stock')
  final double remainingStock;
  final double cost;
  @ColumnInfo(name: 'is_synced')
  final bool isSynced;

  BatchEntity({
    required this.id,
    required this.insumoId,
    required this.batchNumber,
    this.receivedDate,
    required this.expirationDate,
    required this.remainingStock,
    required this.cost,
    this.isSynced = false,
  });
}
